src/components/sbb-file-selector/sbb-file-selector.tsx

```diff
@@ -0,0 +1,296 @@
+    if (this.variant === 'dropzone') {
+      this._dragTarget = dragTarget;
+      toggleDatasetEntry(this._element, 'active', isDragEnter);
+      // FIXME the next line causes button flickering; same if directly setting the button background-color variable
+      toggleDatasetEntry(this._loadButton, 'active', isDragEnter);
+    }
+  }
+
+  private _readFiles(event): void {
+    try {
+      if (event.target.files) {
+        this._createFilesList(event.target.files);
+      }
+    } catch (e) {
+      this.error.emit(e);
+    }
+  }
+
+  private _createFilesList(files: File[]): void {
+    if (
+      !this.multiple ||
+      this.multipleMode !== 'persistent' ||
+      !this.files ||
+      this.files.length === 0
+    ) {
+      this.files = Array.from(files);
+    } else {
+      this.files = Array.from(files)
+        .filter(
+          (newFile: File) =>
+            this.files.findIndex((oldFile: File) => this._checkFileEquality(newFile, oldFile)) ===
+            -1,
+        )
+        .concat(this.files);
+    }
+    this.fileChangedEvent.emit(this.files);
+  }
+
+  private _removeFile(file: File): void {
+    this.files = this.files.filter((f: File) => !this._checkFileEquality(file, f));
+    this.fileChangedEvent.emit(this.files);
+  }
+
+  /** Calculates the correct unit for the file's size. */
+  private _formatFileSize(size: number): string {
+    const i: number = Math.floor(Math.log(size) / Math.log(1024));
+    return `${(size / Math.pow(1024, i)).toFixed(0)} ${this._suffixes[i]}`;
+  }
+
+  private _renderDefaultMode(): JSX.Element {
+    return (
+      <sbb-button
+        variant="secondary"
+        size="m"
+        is-static={true}
+        icon-name="folder-open-small"
+        disabled={this.disabled}
+        ref={(el): void => {
+          this._loadButton = el;
+        }}
+      >
+        {i18nFileSelectorButtonLabel[this._currentLanguage]}
+      </sbb-button>
+    );
+  }
+
+  private _renderDropzoneArea(): JSX.Element {
+    return (
+      <div class="sbb-file-selector__dropzone-area">
+        <div class="sbb-file-selector__dropzone-area--icon">
+          <sbb-icon name="folder-open-medium"></sbb-icon>
+        </div>
+        <div class="sbb-file-selector__dropzone-area--title">{this.titleContent}</div>
+        <div class="sbb-file-selector__dropzone-area--subtitle">
+          {i18nFileSelectorSubtitleLabel[this._currentLanguage]}
+        </div>
+        <div class="sbb-file-selector__dropzone-area--button">
+          <sbb-button
+            variant="secondary"
+            size="m"
+            disabled={this.disabled}
+            is-static={true}
+            ref={(el): void => {
+              this._loadButton = el;
+            }}
+          >
+            {i18nFileSelectorButtonLabel[this._currentLanguage]}
+          </sbb-button>
+        </div>
+      </div>
+    );
+  }
+
+  private _renderFileList(): JSX.Element {
+    return (
+      <ul class="sbb-file-selector__file-list" role="list">
```

Comment:
```suggestion
      <ul class="sbb-file-selector__file-list">
```
should have implicit role list

Reply:
Fixed in [58e6011](https://github.com/lyne-design-system/lyne-components/pull/1894/commits/58e6011496b77e847dde243ffed0a60e45915757)